ts/features/bonus/cgn/components/merchants/CgnDiscountDetail.tsx

```diff
@@ -0,0 +1,147 @@
+import * as React from "react";
+import { View } from "native-base";
+import { StyleSheet } from "react-native";
+import { TouchableWithoutFeedback } from "@gorhom/bottom-sheet";
+import { TmpDiscountType } from "../../__mock__/availableMerchantDetail";
+import { useIOBottomSheet } from "../../../../../utils/bottomSheet";
+import I18n from "../../../../../i18n";
+import { IOStyles } from "../../../../../components/core/variables/IOStyles";
+import { H3 } from "../../../../../components/core/typography/H3";
+import { H5 } from "../../../../../components/core/typography/H5";
+import { H4 } from "../../../../../components/core/typography/H4";
+import IconFont from "../../../../../components/ui/IconFont";
+import { IOColors } from "../../../../../components/core/variables/IOColors";
+import { clipboardSetStringWithFeedback } from "../../../../../utils/clipboard";
+import { BaseTypography } from "../../../../../components/core/typography/BaseTypography";
+
+type Props = {
+  discount: TmpDiscountType;
+};
+
+const styles = StyleSheet.create({
+  row: {
+    flexDirection: "row"
+  },
+  verticalPadding: {
+    paddingBottom: 16
+  },
+  discountValueBox: {
+    borderRadius: 6.5,
+    paddingVertical: 5,
+    width: 40,
+    textAlign: "center",
+    backgroundColor: "#EB9505"
+  },
+  container: {
+    paddingTop: 16
+  },
+  codeContainer: { alignItems: "center", justifyContent: "space-between" },
+  codeText: {
+    fontSize: 20
+  },
+  flexEnd: { alignSelf: "flex-end" },
+  discountValue: { textAlign: "center", lineHeight: 30 }
+});
+
+const PERCENTAGE_SYMBOL = "%";
```

Comment:
Since it used in few places could it a be shared const defined once a time?

Reply:
Since the Box is the same in both components unless it's dimension i recreated a shared component which handles both dimensions

you can check fb2889e